<launch>
      
    <node pkg="camera1394stereo" type="camera1394stereo_node" name="camera1394stereo_node" output="screen" >
	    <param name="video_mode" value="format7_mode3" />
	    <param name="format7_color_coding" value="raw16" />
	    <param name="bayer_pattern" value="bggr" />
	    <param name="bayer_method" value="" />
	    <param name="stereo_method" value="Interlaced" />
	    <param name="camera_info_url_left" value="" />
	    <param name="camera_info_url_right" value="" />
  	</node>
  	
  	<arg name="pi/2" value="1.5707963267948966" />
    <arg name="optical_rotate" value="0 0 0 -$(arg pi/2) 0 -$(arg pi/2)" />
  	<node pkg="tf" type="static_transform_publisher" name="camera_base_link"
        args="$(arg optical_rotate) base_link stereo_camera 100" />  
  	
  	<!-- Run the ROS package stereo_image_proc -->
	<group ns="/stereo_camera" >
		<node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc"/>
		
		<!-- Odometry -->
		<node pkg="rtabmap_ros" type="stereo_odometry" name="stereo_odometry" output="screen">
	  	  	<remap from="left/image_rect"       to="left/image_rect"/>
	  	  	<remap from="right/image_rect"      to="right/image_rect"/>
	  	  	<remap from="left/camera_info"      to="left/camera_info"/>
	  	  	<remap from="right/camera_info"     to="right/camera_info"/>
		  	<remap from="odom" to="odom_stereo"/>
	    
	  	  	<param name="frame_id" type="string" value="base_link"/>
	  	  	<param name="odom_frame_id" type="string" value="odom"/>
	  	  	<param name="approx_sync" type="bool" value="false"/>
	  	  	<param name="queue_size" type="int" value="5"/>

			<param name="Odom/InlierDistance" type="string" value="0.1"/>
	  		<param name="Odom/MinInliers" type="string" value="10"/>
	  		<param name="Odom/RoiRatios" type="string" value="0.03 0.03 0.04 0.04"/>
	  		<param name="Odom/MaxDepth" type="string" value="10"/>
      
	  		<param name="GFTT/MaxCorners" type="string" value="500"/>
	  		<param name="GFTT/MinDistance" type="string" value="5"/>
	  	</node>		
	</group>
  
  
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rtabmap_ros)/launch/config/test_odometry.rviz"/>
  
</launch>